```
Undefined symbols for architecture x86_64:
  "_OBJC_CLASS_$_NSString", referenced from:
      objc-class-ref in filepathurl.o
  "_objc_msgSend", referenced from:
      stringFromNSString(NSString const*) in filepathurl.o
      urlFromNSURL(NSURL const*) in filepathurl.o
      ensureFilePathUrl(QUrl const&) in filepathurl.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

I guess I am missing a linker command.
Who knows how to solve this?

--- a/build/depends.py
+++ b/build/depends.py
@@ -24,8 +24,7 @@ class PortAudio(Dependence):
     def sources(self, build):
         return ['soundio/sounddeviceportaudio.cpp']
         
-        
-class FilePathURL(Dependence):
+class OSXFilePathUrlBackport(Dependence):
 
     def configure(self, build, conf):
         return
@@ -1538,7 +1537,7 @@ class MixxxCore(Feature):
         return [SoundTouch, ReplayGain, Ebur128Mit, PortAudio, PortMIDI, Qt, TestHeaders,
                 FidLib, SndFile, FLAC, OggVorbis, OpenGL, TagLib, ProtoBuf,
                 Chromaprint, RubberBand, SecurityFramework, CoreServices, Foundation, IOKit,
-                QtScriptByteArray, Reverb, FpClassify, PortAudioRingBuffer, FilePathURL]
+                QtScriptByteArray, Reverb, FpClassify, PortAudioRingBuffer, OSXFilePathUrlBackport]
 
     def post_dependency_check_configure(self, build, conf):
         """Sets up additional things in the Environment that must happen

--- a/build/depends.py
+++ b/build/depends.py
@@ -105,6 +105,13 @@ class CoreServices(Dependence):
         build.env.Append(CPPPATH='/System/Library/Frameworks/CoreServices.framework/Headers/')
         build.env.Append(LINKFLAGS='-framework CoreServices')
 
+class Foundation(Dependence):
+    def configure(self, build, conf):
+        if not build.platform_is_osx:
+            return
+        build.env.Append(CPPPATH='/System/Library/Frameworks/Foundation.framework/Headers/')
+        build.env.Append(LINKFLAGS='-framework Foundation')
+
 class IOKit(Dependence):
     """Used for battery measurements and controlling the screensaver on OS X and iOS."""
     def configure(self, build, conf):
@@ -1530,7 +1537,7 @@ class MixxxCore(Feature):
     def depends(self, build):
         return [SoundTouch, ReplayGain, Ebur128Mit, PortAudio, PortMIDI, Qt, TestHeaders,
                 FidLib, SndFile, FLAC, OggVorbis, OpenGL, TagLib, ProtoBuf,
-                Chromaprint, RubberBand, SecurityFramework, CoreServices, IOKit,
+                Chromaprint, RubberBand, SecurityFramework, CoreServices, Foundation, IOKit,
                 QtScriptByteArray, Reverb, FpClassify, PortAudioRingBuffer, FilePathURL]
 
     def post_dependency_check_configure(self, build, conf):

--- a/src/util/filepathurl.mm
+++ b/src/util/filepathurl.mm
@@ -1,11 +1,6 @@
 #include "util/filepathurl.h"
 
-#include <QtCore/qsystemdetection.h>
-#if defined(Q_OS_IOS)
-#import <UIKit/UIKit.h>
-#elif defined(Q_OS_OSX)
-#import <Cocoa/Cocoa.h>
-#endif
+
 
 #import <Foundation/Foundation.h>
 
@@ -36,6 +31,5 @@ QUrl ensureFilePathUrl(const QUrl& url) {
     if ([nsurl isFileReferenceURL]) {
         return urlFromNSURL([nsurl filePathURL]);
     }
-    return url; 
-} 
-
+    return url;
+}

--- a/src/util/filepathurl.mm
+++ b/src/util/filepathurl.mm
@@ -4,8 +4,10 @@
 
 #import <Foundation/Foundation.h>
 
-
-// Taken from src/platformsupport/clipboard/qmacmime.mm
+// This File uses Apples Foundation Kit to converte a QUrl containing a file 
+// reference QUrl("file:///.file/id=6571367.1629051") to a normal file path 
+// QUrl Mixxx can handle. 
+// Taken from https://github.com/qt/qtbase/blob/5.11/src/platformsupport/clipboard/qmacmime.mm#L622
 
 QString stringFromNSString(const NSString *string) {
     if (!string)

